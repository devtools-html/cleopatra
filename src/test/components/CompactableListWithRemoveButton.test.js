/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

// @flow
import React from 'react';
import ReactDOM from 'react-dom';
import CompactableListWithRemoveButton from '../../components/shared/CompactableListWithRemoveButton';
import renderer from 'react-test-renderer';

// For the react-transition-group component
ReactDOM.findDOMNode = jest.fn(() => ({ className: '' }));

describe('shared/CompactableListWithRemoveButton', function() {
  const introductions = ['', 'Some introduction'];
  introductions.forEach(showIntroduction => {
    describe(`with showIntroduction set to '${showIntroduction}'`, function() {
      it('renders compact and full lists with various elements', function() {
        const list = renderer.create(
          <CompactableListWithRemoveButton
            className="some-class"
            compact={true}
            items={['foo', 'bar']}
            buttonTitle="button title"
            showIntroduction={showIntroduction}
            onItemRemove={() => {}}
          />
        );
        expect(list).toMatchSnapshot();

        list.update(
          <CompactableListWithRemoveButton
            className="some-class"
            compact={false}
            items={['foo', 'bar']}
            buttonTitle="button title"
            showIntroduction={showIntroduction}
            onItemRemove={() => {}}
          />
        );
        expect(list).toMatchSnapshot();

        list.update(
          <CompactableListWithRemoveButton
            className="some-class"
            compact={false}
            items={['foo', 'bar', 'hello']}
            buttonTitle="button title"
            showIntroduction={showIntroduction}
            onItemRemove={() => {}}
          />
        );
        expect(list).toMatchSnapshot();

        list.update(
          <CompactableListWithRemoveButton
            className="some-class"
            compact={false}
            items={[]}
            buttonTitle="button title"
            showIntroduction={showIntroduction}
            onItemRemove={() => {}}
          />
        );
        expect(list).toMatchSnapshot();

        list.update(
          <CompactableListWithRemoveButton
            className="some-class"
            compact={false}
            items={[]}
            buttonTitle="button title"
            showIntroduction={showIntroduction}
            onItemRemove={() => {}}
          />
        );
        expect(list).toMatchSnapshot();
      });
    });
  });
});
