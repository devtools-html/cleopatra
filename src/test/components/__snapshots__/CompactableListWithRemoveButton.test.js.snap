// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shared/CompactableListWithRemoveButton with showIntroduction set to '' renders compact and full lists with various elements 1`] = `
<div
  className="listWithRemoveButton some-class listWithRemoveButton_isCompact"
>
  <div
    className="listWithRemoveButton_compact"
  >
    <div
      className="listWithRemoveButton_compact-itemList"
    >
      <span>
        foo
      </span>
       and 
      <span>
        bar
      </span>
    </div>
    <div
      className="listWithRemoveButton_compact-itemCount"
    >
      (
      2
      )
    </div>
  </div>
  <div
    className="listWithRemoveButton_full"
  >
    <div
      className="listWithRemoveButton_full-itemBox"
    >
      <div
        className="listWithRemoveButton_full-item"
      >
        foo
      </div>
      <button
        aria-label="button title"
        className="listWithRemoveButton_full-button"
        data-index={0}
        onMouseDown={[Function]}
        title="button title"
        type="button"
      >
        ✕
      </button>
    </div>
    <div
      className="listWithRemoveButton_full-itemBox"
    >
      <div
        className="listWithRemoveButton_full-item"
      >
        bar
      </div>
      <button
        aria-label="button title"
        className="listWithRemoveButton_full-button"
        data-index={1}
        onMouseDown={[Function]}
        title="button title"
        type="button"
      >
        ✕
      </button>
    </div>
  </div>
</div>
`;

exports[`shared/CompactableListWithRemoveButton with showIntroduction set to '' renders compact and full lists with various elements 2`] = `
<div
  className="listWithRemoveButton some-class listWithRemoveButton_isFull"
>
  <div
    className="listWithRemoveButton_compact"
  >
    <div
      className="listWithRemoveButton_compact-itemList"
    >
      <span>
        foo
      </span>
       and 
      <span>
        bar
      </span>
    </div>
    <div
      className="listWithRemoveButton_compact-itemCount"
    >
      (
      2
      )
    </div>
  </div>
  <div
    className="listWithRemoveButton_full"
  >
    <div
      className="listWithRemoveButton_full-itemBox"
    >
      <div
        className="listWithRemoveButton_full-item"
      >
        foo
      </div>
      <button
        aria-label="button title"
        className="listWithRemoveButton_full-button"
        data-index={0}
        onMouseDown={[Function]}
        title="button title"
        type="button"
      >
        ✕
      </button>
    </div>
    <div
      className="listWithRemoveButton_full-itemBox"
    >
      <div
        className="listWithRemoveButton_full-item"
      >
        bar
      </div>
      <button
        aria-label="button title"
        className="listWithRemoveButton_full-button"
        data-index={1}
        onMouseDown={[Function]}
        title="button title"
        type="button"
      >
        ✕
      </button>
    </div>
  </div>
</div>
`;

exports[`shared/CompactableListWithRemoveButton with showIntroduction set to '' renders compact and full lists with various elements 3`] = `
<div
  className="listWithRemoveButton some-class listWithRemoveButton_isFull"
>
  <div
    className="listWithRemoveButton_compact"
  >
    <div
      className="listWithRemoveButton_compact-itemList"
    >
      <span>
        foo
      </span>
      , 
      <span>
        bar
      </span>
       and 
      <span>
        hello
      </span>
    </div>
    <div
      className="listWithRemoveButton_compact-itemCount"
    >
      (
      3
      )
    </div>
  </div>
  <div
    className="listWithRemoveButton_full"
  >
    <div
      className="listWithRemoveButton_full-itemBox"
    >
      <div
        className="listWithRemoveButton_full-item"
      >
        foo
      </div>
      <button
        aria-label="button title"
        className="listWithRemoveButton_full-button"
        data-index={0}
        onMouseDown={[Function]}
        title="button title"
        type="button"
      >
        ✕
      </button>
    </div>
    <div
      className="listWithRemoveButton_full-itemBox"
    >
      <div
        className="listWithRemoveButton_full-item"
      >
        bar
      </div>
      <button
        aria-label="button title"
        className="listWithRemoveButton_full-button"
        data-index={1}
        onMouseDown={[Function]}
        title="button title"
        type="button"
      >
        ✕
      </button>
    </div>
    <div
      className="listWithRemoveButton_full-itemBox"
    >
      <div
        className="listWithRemoveButton_full-item"
      >
        hello
      </div>
      <button
        aria-label="button title"
        className="listWithRemoveButton_full-button"
        data-index={2}
        onMouseDown={[Function]}
        title="button title"
        type="button"
      >
        ✕
      </button>
    </div>
  </div>
</div>
`;

exports[`shared/CompactableListWithRemoveButton with showIntroduction set to '' renders compact and full lists with various elements 4`] = `
<div
  className="listWithRemoveButton some-class listWithRemoveButton_isFull"
>
  <div
    className="listWithRemoveButton_full"
  >
    <div
      className="listWithRemoveButton_full-itemBox"
    >
      <div
        className="listWithRemoveButton_full-item"
      >
        foo
      </div>
      <button
        aria-label="button title"
        className="listWithRemoveButton_full-button"
        data-index={0}
        onMouseDown={[Function]}
        title="button title"
        type="button"
      >
        ✕
      </button>
    </div>
    <div
      className="listWithRemoveButton_full-itemBox"
    >
      <div
        className="listWithRemoveButton_full-item"
      >
        bar
      </div>
      <button
        aria-label="button title"
        className="listWithRemoveButton_full-button"
        data-index={1}
        onMouseDown={[Function]}
        title="button title"
        type="button"
      >
        ✕
      </button>
    </div>
    <div
      className="listWithRemoveButton_full-itemBox"
    >
      <div
        className="listWithRemoveButton_full-item"
      >
        hello
      </div>
      <button
        aria-label="button title"
        className="listWithRemoveButton_full-button"
        data-index={2}
        onMouseDown={[Function]}
        title="button title"
        type="button"
      >
        ✕
      </button>
    </div>
  </div>
</div>
`;

exports[`shared/CompactableListWithRemoveButton with showIntroduction set to '' renders compact and full lists with various elements 5`] = `
<div
  className="listWithRemoveButton some-class listWithRemoveButton_isFull"
>
  <div
    className="listWithRemoveButton_full"
  >
    <div
      className="listWithRemoveButton_full-itemBox"
    >
      <div
        className="listWithRemoveButton_full-item"
      >
        foo
      </div>
      <button
        aria-label="button title"
        className="listWithRemoveButton_full-button"
        data-index={0}
        onMouseDown={[Function]}
        title="button title"
        type="button"
      >
        ✕
      </button>
    </div>
    <div
      className="listWithRemoveButton_full-itemBox"
    >
      <div
        className="listWithRemoveButton_full-item"
      >
        bar
      </div>
      <button
        aria-label="button title"
        className="listWithRemoveButton_full-button"
        data-index={1}
        onMouseDown={[Function]}
        title="button title"
        type="button"
      >
        ✕
      </button>
    </div>
    <div
      className="listWithRemoveButton_full-itemBox"
    >
      <div
        className="listWithRemoveButton_full-item"
      >
        hello
      </div>
      <button
        aria-label="button title"
        className="listWithRemoveButton_full-button"
        data-index={2}
        onMouseDown={[Function]}
        title="button title"
        type="button"
      >
        ✕
      </button>
    </div>
  </div>
</div>
`;

exports[`shared/CompactableListWithRemoveButton with showIntroduction set to 'Some introduction' renders compact and full lists with various elements 1`] = `
<div
  className="listWithRemoveButton some-class listWithRemoveButton_isCompact"
>
  <div
    className="listWithRemoveButton_compact"
  >
    <div
      className="listWithRemoveButton_compact-itemList"
    >
      <span>
        foo
      </span>
       and 
      <span>
        bar
      </span>
    </div>
    <div
      className="listWithRemoveButton_compact-itemCount"
    >
      (
      2
      )
    </div>
  </div>
  <div
    className="listWithRemoveButton_full"
  >
    <div
      className="listWithRemoveButton_full-itemBox"
    >
      <div
        className="listWithRemoveButton_full-item"
      >
        foo
      </div>
      <button
        aria-label="button title"
        className="listWithRemoveButton_full-button"
        data-index={0}
        onMouseDown={[Function]}
        title="button title"
        type="button"
      >
        ✕
      </button>
    </div>
    <div
      className="listWithRemoveButton_full-itemBox"
    >
      <div
        className="listWithRemoveButton_full-item"
      >
        bar
      </div>
      <button
        aria-label="button title"
        className="listWithRemoveButton_full-button"
        data-index={1}
        onMouseDown={[Function]}
        title="button title"
        type="button"
      >
        ✕
      </button>
    </div>
  </div>
</div>
`;

exports[`shared/CompactableListWithRemoveButton with showIntroduction set to 'Some introduction' renders compact and full lists with various elements 2`] = `
<div
  className="listWithRemoveButton some-class listWithRemoveButton_isFull"
>
  <div
    className="listWithRemoveButton_compact"
  >
    <div
      className="listWithRemoveButton_compact-itemList"
    >
      <span>
        foo
      </span>
       and 
      <span>
        bar
      </span>
    </div>
    <div
      className="listWithRemoveButton_compact-itemCount"
    >
      (
      2
      )
    </div>
  </div>
  <div
    className="listWithRemoveButton_full"
  >
    <div
      className="listWithRemoveButton_full-itemBox"
    >
      <div
        className="listWithRemoveButton_full-item"
      >
        foo
      </div>
      <button
        aria-label="button title"
        className="listWithRemoveButton_full-button"
        data-index={0}
        onMouseDown={[Function]}
        title="button title"
        type="button"
      >
        ✕
      </button>
    </div>
    <div
      className="listWithRemoveButton_full-itemBox"
    >
      <div
        className="listWithRemoveButton_full-item"
      >
        bar
      </div>
      <button
        aria-label="button title"
        className="listWithRemoveButton_full-button"
        data-index={1}
        onMouseDown={[Function]}
        title="button title"
        type="button"
      >
        ✕
      </button>
    </div>
  </div>
</div>
`;

exports[`shared/CompactableListWithRemoveButton with showIntroduction set to 'Some introduction' renders compact and full lists with various elements 3`] = `
<div
  className="listWithRemoveButton some-class listWithRemoveButton_isFull"
>
  <div
    className="listWithRemoveButton_compact"
  >
    <div
      className="listWithRemoveButton_compact-itemList"
    >
      <span>
        foo
      </span>
      , 
      <span>
        bar
      </span>
       and 
      <span>
        hello
      </span>
    </div>
    <div
      className="listWithRemoveButton_compact-itemCount"
    >
      (
      3
      )
    </div>
  </div>
  <div
    className="listWithRemoveButton_full"
  >
    <div
      className="listWithRemoveButton_full-itemBox"
    >
      <div
        className="listWithRemoveButton_full-item"
      >
        foo
      </div>
      <button
        aria-label="button title"
        className="listWithRemoveButton_full-button"
        data-index={0}
        onMouseDown={[Function]}
        title="button title"
        type="button"
      >
        ✕
      </button>
    </div>
    <div
      className="listWithRemoveButton_full-itemBox"
    >
      <div
        className="listWithRemoveButton_full-item"
      >
        bar
      </div>
      <button
        aria-label="button title"
        className="listWithRemoveButton_full-button"
        data-index={1}
        onMouseDown={[Function]}
        title="button title"
        type="button"
      >
        ✕
      </button>
    </div>
    <div
      className="listWithRemoveButton_full-itemBox"
    >
      <div
        className="listWithRemoveButton_full-item"
      >
        hello
      </div>
      <button
        aria-label="button title"
        className="listWithRemoveButton_full-button"
        data-index={2}
        onMouseDown={[Function]}
        title="button title"
        type="button"
      >
        ✕
      </button>
    </div>
  </div>
</div>
`;

exports[`shared/CompactableListWithRemoveButton with showIntroduction set to 'Some introduction' renders compact and full lists with various elements 4`] = `
<div
  className="listWithRemoveButton some-class listWithRemoveButton_isFull"
>
  <div
    className="listWithRemoveButton_introduction"
  >
    Some introduction
  </div>
</div>
`;

exports[`shared/CompactableListWithRemoveButton with showIntroduction set to 'Some introduction' renders compact and full lists with various elements 5`] = `
<div
  className="listWithRemoveButton some-class listWithRemoveButton_isFull"
>
  <div
    className="listWithRemoveButton_introduction"
  >
    Some introduction
  </div>
</div>
`;
